test/integration-test/TestIIRDeserializer.cpp

```diff
@@ -0,0 +1,598 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _                      
+//                         | |                     
+//                       __| | __ ___      ___ ___  
+//                      / _` |/ _` \ \ /\ / / '_  | 
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT). 
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/IIR/ASTStmt.h"
+#include "dawn/IIR/ASTUtil.h"
+#include "dawn/IIR/FieldAccessMetadata.h"
+#include "dawn/IIR/IIRNodeIterator.h"
+#include "dawn/IIR/InstantiationHelper.h"
+#include "dawn/IIR/StencilInstantiation.h"
+#include "dawn/Serialization/IIRSerializer.h"
+#include "dawn/Optimizer/AccessComputation.h"
+#include "dawn/Optimizer/PassComputeStageExtents.h"
+#include "dawn/Optimizer/PassTemporaryType.h"
+#include "dawn/Optimizer/PassSetStageName.h"
+#include "dawn/Optimizer/StatementMapper.h"
+#include "dawn/SIR/ASTFwd.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/Support/Logging.h"
+#include "dawn/Support/STLExtras.h"
+#include "dawn/IIR/IIR.h"
+#include "dawn/IIR/StencilMetaInformation.h"
+
+#include <stack>
+#include <memory>
+#include <string>
+#include <unistd.h>
+#include <gtest/gtest.h>
+
+using namespace dawn;
+
+namespace {
+
+void compareIIRstructures(iir::IIR* lhs, iir::IIR* rhs) {
+  ASSERT_TRUE(lhs->checkTreeConsistency());
```

Comment:
It would be cool if all of this could be implemented as a Visitor, but I don't think that works ... :-)